### ISSUE TITLE
stash apply: report merge conflicts

### ISSUE THREAD

**Issue by josmith42** [extrawurst/gitui#576]
**Is your feature request related to a problem? Please describe.**
When applying a stash that will result in conflict (e.g., the working copy has changed), gitui does not report the error message. This makes it difficult to see that a conflict occurred.

**Describe the solution you'd like**
A dialog box with the error message. (e.g.: "CONFLICT (content): Merge conflict in [filename]", maybe with an OK button and an Abort button.

**Describe alternatives you've considered**
The error message could also be in the status bar.

**Additional context**
If there is already error handling for stashes, maybe it could be reused.

**Comment by josmith42** [comment]
No, it appears to not understand the conflict state after a stash is applied:
![gutui-stash-apply-conflict](https://user-images.githubusercontent.com/841270/110458740-1d4f7e80-809a-11eb-8e50-1a53eee8e5b7.png)
This occurs even if the stash is applied outside of gitui.

Whereas, if you do a `git status`, you see the following:
```
On branch master
Unmerged paths:
  (use "git restore --staged <file>..." to unstage)
  (use "git add <file>..." to mark resolution)
	both modified:   conflict

no changes added to commit (use "git add" and/or "git commit -a")
```

**Comment by extrawurst** [comment]
which gitui version are you using?

**Comment by josmith42** [comment]
I _was_ using 0.11.0. However, I just upgraded to 0.12.0 and it still occurs.

**Comment by extrawurst** [comment]
0.12 should tell you that the repo is in a conflicting state though 🤔

**Comment by josmith42** [comment]
It just shows `M` in Unstaged and Stages, the same as the above screenshot. :-(

**Comment by extrawurst** [comment]
@josmith42 ok I am having trouble reproducing this. applying that results in a conflict actually is prevented 
<img width="1072" alt="image" src="https://user-images.githubusercontent.com/776816/110470109-bd2cfc80-80da-11eb-80a0-24143a41c0ce.png">

**Comment by josmith42** [comment]
OK, here are some steps to repro:
* Make a change to a file. Stash the change.
* Make a different change to the same file on the same line. Commit the change.
* Use gitui to apply the stash you saved in the first step.

So after #578, gitui does show the files as conflicted, but it still doesn't prevent the confilct from occurring.

To me, showing the confilct is sufficient, but if if you want gitui to prevent the conflict in the first place, it is not doing that in this case.

**Comment by josmith42** [comment]
Github won't allow me to reopen this issue, so please reopen the issue if you feel it should be based on what I reported above.

**Comment by extrawurst** [comment]
comparing this again against regular `git stash apply` it seems to be the default behaviour to put the files into a conflicting state. meaning I will leave it like that. the only thing I am going to check now is why the diff is empty for the conflicts

**Comment by extrawurst** [comment]
superseded by https://github.com/extrawurst/gitui/issues/580
